### PR TITLE
Add StackExchange.Redis 2.X.X package

### PR DESCRIPTION
**Pull request by BrennanConroy**
Some of this can be cleaned up in the 3.0 change.

Basically copied all the SignalR.Redis and SignalR.Redis.Tests code into new folders and moved the TestConnectionMulitplexer file into both Test folders so the Test.Utils project wouldn't cause issues.

And skipped/removed all the unit tests for StackExchangeRedis.Tests which is tracked by https://github.com/aspnet/SignalR/issues/3088

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -62,6 +62,7 @@
     <NETStandardLibrary20PackageVersion>2.0.3</NETStandardLibrary20PackageVersion>
     <NewtonsoftJsonPackageVersion>11.0.2</NewtonsoftJsonPackageVersion>
     <StackExchangeRedisStrongNamePackageVersion>1.2.6</StackExchangeRedisStrongNamePackageVersion>
+    <StackExchangeRedisPackageVersion>2.0.513</StackExchangeRedisPackageVersion>
```
Comment (BrennanConroy) [inline build/dependencies.props]: @Eilon For dependency

**Comment by BrennanConroy** [comment]
Dang, guess I have to disable the Redis benchmarks for 2.2

**Comment by anurse** [comment]
As discussed in person, can you try to share as much of the code as possible? If only so the code reviewing is easier :).

File: test/Microsoft.AspNetCore.SignalR.StackExchangeRedis.Tests/Docker.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.AspNetCore.SignalR.Redis.Tests
+{
+    public class Docker
```
[the remark this answers is not in the document]
Comment (BrennanConroy) [reply test/Microsoft.AspNetCore.SignalR.StackExchangeRedis.Tests/Docker.cs]: They aren't identical, was trying to make both redis functional tests work side-by-side by changing ports and container names

File: src/Microsoft.AspNetCore.SignalR.StackExchangeRedis/RedisHubLifetimeManager.cs [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR.Redis.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using StackExchange.Redis;
+
+namespace Microsoft.AspNetCore.SignalR.Redis
```
[the remark this answers is not in the document]
Comment (BrennanConroy) [reply src/Microsoft.AspNetCore.SignalR.StackExchangeRedis/RedisHubLifetimeManager.cs]: That's a breaking change, and it causes us to not be able to share any test code

[the remark this answers is not in the document]
Comment (anurse) [reply src/Microsoft.AspNetCore.SignalR.StackExchangeRedis/RedisHubLifetimeManager.cs]: We can't ship these with the same namespace, it's weird. Why is it breaking? This is a brand new package? Only change it for the new code, using an `#if`. It's awkward but it keeps things in one file (good for reviewing) and it'll go away in 3.0

[the remark this answers is not in the document]
Comment (BrennanConroy) [reply src/Microsoft.AspNetCore.SignalR.StackExchangeRedis/RedisHubLifetimeManager.cs]: Its a "new" package, users are supposed to be able to drop the other package and put this one in and not change anything.

[the remark this answers is not in the document]
Comment (anurse) [reply src/Microsoft.AspNetCore.SignalR.StackExchangeRedis/RedisHubLifetimeManager.cs]: > and not change anything

Who said that? I don't think it's supposed to be a drop-in at all. In fact, I think we still need to close on what the extension method is (is it `AddRedis` or `AddStackExchangeRedis`?)

File: src/Microsoft.AspNetCore.SignalR.StackExchangeRedis/Microsoft.AspNetCore.SignalR.StackExchangeRedis.csproj [hunk 4]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Description>Redis for ASP.NET Core SignalR.</Description>
```
Comment (anurse) [inline src/Microsoft.AspNetCore.SignalR.StackExchangeRedis/Microsoft.AspNetCore.SignalR.StackExchangeRedis.csproj]: Let's include the library name here too so it's clear. Suggest:

```
Provides scale-out support for ASP.NET Core using a Redis server and the StackExchange.Redis client.
```

**Comment by buvinghausen** [comment]
Many thanks @BrennanConroy

**Pull request merged**